File: src/test/java/cn/enaium/cf4m/test/ConfigurationTest.java
package cn.enaium.cf4m.test;

import cn.enaium.cf4m.annotation.Configuration;
import cn.enaium.cf4m.configuration.ICommandConfiguration;
import cn.enaium.cf4m.configuration.IConfiguration;

/**
 * @author Enaium
 */
@Configuration
public class ConfigurationTest implements IConfiguration {
    @Override
    public ICommandConfiguration getCommand() {
        return new ICommandConfiguration() {
            @Override
            public String getPrefix() {
                return "-";
            }
        };
    }
}

File: src/test/java/cn/enaium/cf4m/test/ConfigurationTest.kt
package cn.enaium.cf4m.test

import cn.enaium.cf4m.annotation.Configuration
import cn.enaium.cf4m.configuration.ICommandConfiguration
import cn.enaium.cf4m.configuration.IConfiguration
import cn.enaium.cf4m.configuration.IModuleConfiguration
import cn.enaium.cf4m.provider.ModuleProvider

/**
 * @author Enaium
 */
@Configuration
class ConfigurationTest : IConfiguration {
    override fun getCommand(): ICommandConfiguration {
        return object : ICommandConfiguration {
            override fun getPrefix(): String {
                return "-"
            }
        }
    }

    override fun getModule(): IModuleConfiguration {
        return object : IModuleConfiguration {
            override fun enable(module: ModuleProvider) {
                println(module.name + " Enable")
            }

            override fun disable(module: ModuleProvider) {
                println(module.name + " Disable")
            }
        }
    }
}